presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonOperators.java

```diff
@@ -255,7 +255,7 @@ public static Slice castFromDouble(@SqlType(DOUBLE) double value) throws IOExcep
             return dynamicSliceOutput.slice();
         }
         catch (IOException e) {
-            throw new PrestoException(INVALID_CAST_ARGUMENT, format("Cannot cast '%d' to %s", value, JSON));
+            throw new PrestoException(INVALID_CAST_ARGUMENT, format("Cannot cast '%f' to %s", value, JSON));
```

Comment:
Make this %s too. It works for any type and is fine here since we don't need fancy formatting.